### PR TITLE
Fable 5 test: reorganize test files and consolidate generator modules

### PR DESCRIPTION
**Pull request by phalt**
# Change summary

This PR reorganizes the test suite structure and consolidates generator modules to improve code organization and maintainability.

### Test Reorganization
- **Consolidated coverage tests**: Merged multiple `*_coverage.py` test files into their corresponding main test files:
  - `test_generator_coverage.py` → `tests/generators/test_cicerone_compat.py`
  - `test_utils_coverage.py` → `tests/generators/shared/test_utils.py`
  - `test_cli_coverage.py` and `test_cli_full_coverage.py` → `tests/test_cli.py`
  - `test_additional_coverage.py` → Distributed to appropriate test modules
  - `test_edge_case_coverage.py` → `tests/test_cli.py`

- **Reorganized generator tests**: 
  - Moved `tests/generators/framework/test_framework_generator_*.py` → `tests/generators/api/test_generator.py`
  - Created `tests/generators/basic/test_generator.py` for BasicGenerator tests
  - Reorganized test structure to match generator module hierarchy

- **Renamed test files for clarity**:
  - `tests/api/test_requests.py` → `tests/api/test_request_context.py`
  - `tests/generators/api/test_clients.py` (refactored)
  - `tests/generators/shared/test_schemas.py` (refactored)

### Module Consolidation
- **Moved `schema_utils.py` functions**: Consolidated `build_union_type_string()` from `clientele/generators/schema_utils.py` into `clientele/generators/shared/schemas.py`
- **Moved utility functions**: Moved `get_client_project_directory_path()` from `clientele/utils.py` to `clientele/generators/shared/utils.py`
- **Removed intermediate modules**:
  - Deleted `clientele/generators/api/generators/schemas.py` (re-export module)
  - Deleted `clientele/generators/api/generators/clients.py` (unused framework-specific code)
  - Deleted `clientele/utils.py` (functions moved to appropriate locations)
  - Deleted `clientele/generators/schema_utils.py` (functions consolidated)

### Import Updates
- Updated all imports to reflect new module locations:
  - `from clientele.api import requests` → `from clientele.api import request_context`
  - `from clientele.generators.shared.generators.schemas import SchemasGenerator` → `from clientele.generators.shared.schemas import SchemasGenerator`
  - `from clientele.generators.api.generators import clients` → `from clientele.generators.api import clients`

### Code Quality Improvements
- **Enhanced `utils.py`**: Added `get_client_project_directory_path()` function with improved path handling
- **Simplified `settings.py`**: Replaced custom `split_ver()` function with direct use of `sys.version_info`
- **Improved test coverage**: Added comprehensive tests for edge cases in utility functions (empty strings, special characters, digit-starting identifiers, etc.)

## Related issue(s)

N/A

## Pull request tasks

The following have been completed for this task:

- [x] **Code changes**
- [x] **Tests written**
- [x] **Documentation changes for new or changed features** (N/A)
- [x] **Pre-review preparation**
  - [x] `make test` (all tests pass)
  - [x] `make format`
  - [x] `make ty`
  - [x] `make generate-test-clients`
    - [x] Test clients remain the same
- [ ] Changelog updated
- [ ] Contributors updated

## Other information

This refactoring improves code organization by:
1. Eliminating redundant test files with overlapping coverage
2. Consolidating related functionality into single modules
3. Removing intermediate re-export modules
4. Establishing a clearer test structure that mirrors the source code organization
5. Reducing overall code du

https://claude.ai/code/session_012AGyWgJj3Mp5aNFJ5qAB4s

### CODE REVIEW

File: clientele/settings.py [hunk 1]
```diff
@@ -1,10 +1,5 @@
-import platform
+import sys
```
Comment (phalt) [inline clientele/settings.py:1]: This is a legitimately better change.

File: clientele/generators/api/generators/clients.py [hunk 2]
```diff
@@ -1,35 +1,62 @@
+"""Generates the client.py file for the api generator."""
```
Comment (phalt) [inline clientele/generators/api/generators/clients.py:1]: Left over from the recent refactor - this sort of consolidation makes total sense.

File: clientele/api/client.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
```
Comment (phalt) [inline clientele/api/client.py:11]: A genuinely sensible change.

File: tests/test_cli.py [hunk 4]
```diff
@@ -101,17 +101,37 @@ def test_load_openapi_spec_requires_url_or_file():
         cli._load_openapi_spec()
```
Comment (phalt) [inline tests/test_cli.py:101]: Previous iterations of models to generate tests for test coverage naively did so with a new file - this shows how sensible Fable can be at realising it can all be consolidated now (maybe previous models could have done it too - who knows).

**Review by phalt — COMMENTED** [review]
Genuinely useful changes - mostly code reorganisation that I haven't been arsed to do myself, or previous models have failed and got caught in a deep logic loop trying to figure out.
